Fix Upsert config validation to check for metrics aggregation

geeknarrator opened the pull request:
Bugfix for: https://github.com/apache/pinot/issues/8226

Jackie-Jiang reviewed (commented):
I don't think it can compile now because of the extra back brace. Can you please also fix the code format using [Pinot Style](https://docs.pinot.apache.org/developers/developers-and-contributors/code-setup#setup-ide)?

Jackie-Jiang commented on pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/TableConfigUtils.java:
```diff
@@ -534,6 +534,19 @@ static void validateUpsertConfig(TableConfig tableConfig, Schema schema) {
+     static boolean isAggregateMetricsEnabled(TableConfig config) {
+      if(config.getIndexingConfig() == null || config.getIngestionConfig() == null) {return false;
+      }
+      return config.getIndexingConfig().isAggregateMetrics() && CollectionUtils.isEmpty(config.getIngestionConfig().getAggregationConfigs());
```
This should be `||` instead of `&&`

geeknarrator replied:
I think I referred `MutableSegmentImpl` and misunderstood. Fixed it.

geeknarrator commented:
> I don't think it can compile now because of the extra back brace. Can you please also fix the code format using [Pinot Style](https://docs.pinot.apache.org/developers/developers-and-contributors/code-setup#setup-ide)?

Thanks, I configured Pinot style.

geeknarrator commented on pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/TableConfigUtils.java:
```diff
@@ -521,19 +520,32 @@ static void validateUpsertConfig(TableConfig tableConfig, Schema schema) {
+   */
+  @VisibleForTesting
+  static boolean isAggregateMetricsEnabled(TableConfig config) {
+    boolean isAggregateMetricsEnabledInIndexingConfig =
+        Optional.ofNullable(config.getIndexingConfig()).map(IndexingConfig::isAggregateMetrics).orElse(false);
+    boolean hasAggregationConfigs = Optional.ofNullable(config.getIngestionConfig())
+        .map(ingestionConfig -> CollectionUtils.isNotEmpty(ingestionConfig.getAggregationConfigs())).orElse(false);
```
👍 sgtm

geeknarrator commented on pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/TableConfigUtils.java:
```diff
@@ -521,19 +520,32 @@ static void validateUpsertConfig(TableConfig tableConfig, Schema schema) {
+        Optional.ofNullable(config.getIndexingConfig()).map(IndexingConfig::isAggregateMetrics).orElse(false);
+    boolean hasAggregationConfigs = Optional.ofNullable(config.getIngestionConfig())
+        .map(ingestionConfig -> CollectionUtils.isNotEmpty(ingestionConfig.getAggregationConfigs())).orElse(false);
+    return isAggregateMetricsEnabledInIndexingConfig || hasAggregationConfigs;
```
I see.. I didn't know that is a requirement.

geeknarrator replied:
Done.

geeknarrator commented:
@Jackie-Jiang - Thanks for reviewing the changes :) I have addressed the remaining comments. Once you approve those, can I get it merged? I don't seem to have the access.

geeknarrator commented:
@Jackie-Jiang - Could you please help me merge this? I have resolved the conflict. (also the formatting changes are done by the pinot checkstyle configuration.
